pkg/controllers/providers/subnet/controller.go

```diff
@@ -0,0 +1,60 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package subnet
+
+import (
+	"context"
+	"time"
+
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	corecontroller "sigs.k8s.io/karpenter/pkg/operator/controller"
+
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+
+	"github.com/aws/karpenter-provider-aws/pkg/apis/v1beta1"
+	"github.com/aws/karpenter-provider-aws/pkg/providers/subnet"
+)
+
+var _ corecontroller.TypedController[*v1beta1.EC2NodeClass] = (*Controller)(nil)
+
+type Controller struct {
+	subnetProvider subnet.Provider
+}
+
+func NewController(kubeClient client.Client, subnetProvider subnet.Provider) corecontroller.Controller {
+	return corecontroller.Typed[*v1beta1.EC2NodeClass](kubeClient, &Controller{
+		subnetProvider: subnetProvider,
+	})
+}
+
+func (c *Controller) Reconcile(ctx context.Context, nodeClass *v1beta1.EC2NodeClass) (reconcile.Result, error) {
+	return reconcile.Result{RequeueAfter: 5 * time.Minute}, c.subnetProvider.UpdateSubnets(ctx, nodeClass)
```

Comment:
controller-runtime will yell at you if you return both a valid result and an error at the same time